libyul/backends/evm/ControlFlowGraph.h

```diff
@@ -131,6 +131,7 @@ struct CFG
 		std::shared_ptr<DebugData const> debugData;
 		std::reference_wrapper<Scope::Function const> function;
 		std::reference_wrapper<yul::FunctionCall const> functionCall;
+		bool recursive = false;
```

Comment:
Is it only direct? Or indirect too?

Reply:
I guess indirect, too - it's a call, s.t. entering it can end up at the same call again, i.e. a recursive one.
I.e.
```
function f() {
   f() // this call has recursive = true
  g() // this call has recursive = false
  h() // this call has recursive = true
}
function g()
{
  i() // this call has recursive = false - it calls something that's recursive, but the call is not recursive.
}
function h() {
  f() // this call has recursive = true
  g() // this call has recursive = false
}
function i() {
  i() // this call has recursive = true
}
```

Reply:
Right